src/Compiler/Microsoft.AspNetCore.Razor.Language/src/Components/ComponentNodeWriter.cs

```diff
@@ -405,6 +405,12 @@ protected static void UseCapturedCascadingGenericParameterVariable(ComponentInte
         parameter.ReplaceSourceWithCapturedVariable(variableName);
     }
 
+    protected static bool HasDefaultValueExpression(CascadingGenericTypeParameter syntheticArg)
+    {
+        return syntheticArg.ValueExpression is null ||
+            syntheticArg.ValueExpression.StartsWith("default(", StringComparison.Ordinal);
```

Comment:
Are we certain this will never be `default`?

Reply:
Currently, yes, it's only set here:

https://github.com/dotnet/razor/blob/3d25dfb826f9161bf983939032fd0873ac956be8/src/Compiler/Microsoft.AspNetCore.Razor.Language/src/Components/ComponentGenericTypePass.cs#L108

and here (where `variableName` is something like `"__typeInferenceArg_*_*"`):

https://github.com/dotnet/razor/blob/3d25dfb826f9161bf983939032fd0873ac956be8/src/Compiler/Microsoft.AspNetCore.Razor.Language/src/Components/ComponentNodeWriter.cs#L398